src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -170,109 +200,118 @@ private SymbolicRegexMatcher(SymbolicRegexNode<TSet> rootNode, int captureCount,
 
             // Create the initial states for the original pattern.
             var initialStates = new DfaMatchingState<TSet>[statesCount];
-            for (uint i = 0; i < initialStates.Length; i++)
+            for (uint charKind = 0; charKind < initialStates.Length; charKind++)
```

Comment:
Worth a comment that these charKinds map to the CharKind values, that we're relying on 0 being General, etc.